Source/Dafny/Compilers/Compiler-java.cs

```diff
@@ -19,15 +19,33 @@
 using Bpl = Microsoft.Boogie;
 using static Microsoft.Dafny.ConcreteSyntaxTreeUtils;
 
-namespace Microsoft.Dafny {
-  public class JavaCompiler : Compiler {
-    public JavaCompiler(ErrorReporter reporter)
-      : base(reporter) {
+namespace Microsoft.Dafny.Compilers.Java {
+  public class JavaCompiler : SinglePassCompiler {
+    public override void LateInitialize(ErrorReporter reporter, ReadOnlyCollection<string> otherFileNames) {
+      base.LateInitialize(reporter, otherFileNames);
       IntSelect = ",java.math.BigInteger";
       LambdaExecute = ".apply";
     }
 
-    public override String TargetLanguage => "Java";
+    public override IReadOnlySet<string> SupportedExtensions => new HashSet<string> { ".java" };
+
+    public override string TargetLanguage => "Java";
```

Comment:
Should we include the version of java as well? Just a question, no need to change anything.

Reply:
No, I think not; in fact previously these strings were not used at all